src/index.tsx

```diff
@@ -85,3 +85,79 @@ export const useUser = () => {
 
   return context.user
 }
+
+export function useTable(table: string) {
+  const context = React.useContext(SupabaseContext);
+
+
+  if (context === undefined || context === null) {
+    throw new Error('useUser must be used within a SupabaseContext.Provider')
+    return
+  }
+
+  const [data, setData] = React.useState<any[]>([]);
+  const [error, setError] = React.useState<{
+    message: string;
+    details: string;
+    hint: string;
+    code: string;
+  } | null>(null);
+  const [loading, setLoading] = React.useState(true);
+
+  const initialLoad = async () => {
+    const { data: initialData, error: initialError } = await context.sb!
+      .from(table)
+      .select();
```

Comment:
This structure is quite limiting as it doesn't allow to add other methods like .eq() or .limit() to the query. It assumes that you are querying the full table. In many cases, you only want to query and listen for a portion of your table like a single row.

How about a structure like this `const {data, error, loading} = useTable(query, options)`
It could be used like this:
```
const { data, error, loading } = useTable(
  supabase
  .from('products')
  .select('id, name, images, price_range, vendor(name, currency)')
  .eq('id', id),
  { listen: true, types: '*' }
);
  ```

Reply:
It would be great to add a boolean option to listen or not to data update with perhaps what event type you want to listen to. You don't necessarily want to listen to every update. I would put the listener false by default to avoid unnecessary listeners.

Reply:
I'm curious to see how you're thinking about `useRow`. Coming from firebase, I always used firebase-react-hooks either querying a collection or a document so `useTable` and `useRow` kind of replaces that. I quite like to keep it opinionated as well as it simplifies the use but if a use case doesn't fit under these 2 cases, what's the option?

Reply:
I like the approach! It would for sure be possible to add the extra options like `is` and `limit`. I'm just worried a bit about the scale of what needs to be built in order to make this clean and viable.

it almost feels like it's acting as a replacement for the supabase client and that worries me. I see those `useRow` and `useTable` as companions to the main client. In that scenario users wanting to use limit or like would be better served creating the query with the client directly.

One of the reasons I stay want to steer this lib as a companion vs replacement is based on some limitations I had when using `react-firebase-hook` (using two subsequent queries in the same component was a mess to merge errors, loading state and data).  I got some time tonight to work on this so I'll try to add more those options we discussed above :) 

Thanks again for the feedback it makes me think about more than I had seen at first!

Reply:
Make sense. I guess this can always be revisited later on as people use it. useTable and useRow will be quite useful already. Can't wait to play with it.